drv/tdma_client/tdma_client.c

```diff
@@ -0,0 +1,522 @@
+/**
+ * @file
+ * @ingroup     drv_tdma_client
+ *
+ * @brief       Driver for Time-Division-Multiple-Access fot the DotBot radio
+ *
+ * @author Said Alvarado-Marin <said-alexander.alvarado-marin@inria.fr>
+ *
+ * @copyright Inria, 2024
+ */
+#include <nrf.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <string.h>
+#include <stdbool.h>
+
+#include "tdma_client.h"
+#include "clock.h"
+#include "radio.h"
+#include "rng.h"
+#include "timer_hf.h"
+#include "protocol.h"
+#include "device.h"
+#if defined(NRF5340_XXAA) && defined(NRF_NETWORK)
+#include "ipc.h"
+#endif
+
+//=========================== defines ==========================================
+
+#if defined(BOARD_SAILBOT_V1)
+#define TDMA_CLIENT_RADIO_APPLICATION SailBot
+#else
+#define TDMA_CLIENT_RADIO_APPLICATION DotBot
+#endif
+
+#define TDMA_CLIENT_DEFAULT_FRAME_DURATION 500000                              ///< Default duration of the tdma frame, in microseconds.
+#define TDMA_CLIENT_DEFAULT_RX_START       0                                   ///< start of the , in microseconds.
+#define TDMA_CLIENT_DEFAULT_RX_DURATION    TDMA_CLIENT_DEFAULT_FRAME_DURATION  ///< Default duration of the tdma frame, in microseconds.
+#define TDMA_CLIENT_DEFAULT_TX_START       10000                               ///< Default duration of the tdma frame, in microseconds.
+#define TDMA_CLIENT_DEFAULT_TX_DURATION    5000                                ///< Default duration of the tdma frame, in microseconds.
+#define TDMA_CLIENT_HF_TIMER_CC_TX         0                                   ///< Which timer channel will be used for the TX state machine.
+#define TDMA_CLIENT_HF_TIMER_CC_RX         1                                   ///< Which timer channel will be used for the RX state machine.
+#define TDMA_CLIENT_MAX_DELAY_WITHOUT_TX   500000                              ///< Max amount of time that can pass without TXing anything
+#define TDMA_CLIENT_RING_BUFFER_SIZE       10                                  ///< Amount of TX packets the buffer can contain
+#define RADIO_MESSAGE_MAX_SIZE             255                                 ///< Size of buffers used for SPI communications
+#define RADIO_TX_RAMP_UP_TIME              140                                 ///< time it takes the radio to start a transmission
+#define TDMA_CLIENT_TIMER_HF               2
+
+typedef struct {
+    uint8_t  buffer[TDMA_CLIENT_RING_BUFFER_SIZE][PAYLOAD_MAX_LENGTH];  ///< arrays of radio messages waiting to be sent
+    uint32_t packet_length[TDMA_CLIENT_RING_BUFFER_SIZE];               ///< arrays of the lenght of the packets in the buffer
+    uint8_t  writeIndex;                                                ///< Index for next write
+    uint8_t  readIndex;                                                 ///< Index for next read
```

Comment:
Use a consistent naming scheme. This repository uses lower snake case everwhere.
```suggestion
    uint8_t  write_index;                                                ///< Index for next write
    uint8_t  read_index;                                                 ///< Index for next read
```